testcontainers/src/core/containers/request.rs

```diff
@@ -234,27 +244,41 @@ impl PortMapping {
 
 impl<I: Image + Debug> Debug for ContainerRequest<I> {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("ContainerRequest")
-            .field("image", &self.image)
-            .field("overridden_cmd", &self.overridden_cmd)
-            .field("image_name", &self.image_name)
-            .field("image_tag", &self.image_tag)
-            .field("container_name", &self.container_name)
-            .field("network", &self.network)
-            .field("labels", &self.labels)
-            .field("env_vars", &self.env_vars)
-            .field("hosts", &self.hosts)
-            .field("mounts", &self.mounts)
-            .field("ports", &self.ports)
-            .field("ulimits", &self.ulimits)
-            .field("privileged", &self.privileged)
-            .field("cap_add", &self.cap_add)
-            .field("cap_drop", &self.cap_drop)
-            .field("shm_size", &self.shm_size)
-            .field("cgroupns_mode", &self.cgroupns_mode)
-            .field("userns_mode", &self.userns_mode)
-            .field("startup_timeout", &self.startup_timeout)
-            .field("working_dir", &self.working_dir)
-            .finish()
+        let with_feature_flag_fields = {
+            #[cfg(not(feature = "reusable-containers"))]
+            {
+                std::fmt::DebugStruct::finish
+            }
+            #[cfg(feature = "reusable-containers")]
+            {
+                |repr: &mut std::fmt::DebugStruct<'_, '_>| -> std::fmt::Result {
+                    repr.field("reusable", &self.reuse).finish()
+                }
+            }
+        };
+
+        with_feature_flag_fields(
```

Comment:
Tbh, I prefer another pattern for simple use-cases:

```
let debug_struct = f.debug_struct(..); // all common ones are omitted 

// Utilize shadowing if the feature is enabled
#[cfg(feature = "reusable-containers")]
let debug_struct = debug_struct.field(“reusable”, ..);

debug_struct.finish()
```

Reply:
FTR, same. There's a problem with lifetimes in this instance though, and the borrow-checker doesn't seem to be able to be made happy any other way 🥲

![Screenshot 2024-12-16 at 9 32 52 AM](https://github.com/user-attachments/assets/5c1df5ea-c528-402a-b253-c8d5e12e4304)

![Screenshot 2024-12-16 at 9 33 03 AM](https://github.com/user-attachments/assets/71149bd6-e91f-4f0b-933a-1ee8815fe5ce)

Reply:
**_WAIT_**, disregard that - brain wasn't working 🤦. Took a quick reading comprehension lesson and tried again:

![Screenshot 2024-12-16 at 9 50 24 AM](https://github.com/user-attachments/assets/dd4ddf34-e81c-4c12-834f-d70b8fa704db)